src/test/java/uk/gov/pay/api/it/directdebit/AgreementsResourceITest.java

```diff
@@ -95,6 +95,44 @@ public void createPayment_respondsWith500_whenConnectorResponseIsAnUnrecognisedE
                 .extract().body().asString();
     }
 
+    @Test
+    public void shouldGetADirectDebitAgreement() {
```

Comment:
David and I were exploring an idea in the PP-3678-collect-payment branch to test happy path stuff like this with something more akin to a unit test by mocking the service rather than using Wiremock to mock the whole of connector (see `PaymentsResourceCreatePaymentTest.java` in that class).

The idea was we’d end up with unit tests to check the interaction with the service, Pact tests to test the interaction between the service and connector, and integration tests to test the stuff that really does require spinning up the app (such as the conversion of errors to appropriate HTTP responses).

Anyway, the code you’ve written is good is follows the existing conventions on master, so let’s go with it and maybe revisit if what David and I were working on sticks.